src/components/Facility/PatientNotesList.tsx

```diff
@@ -23,57 +24,59 @@ interface PatientNotesProps {
   setReplyTo?: (reply_to: PatientNotesModel | undefined) => void;
 }
 
-const pageSize = RESULTS_PER_PAGE_LIMIT;
-
 const PatientNotesList = (props: PatientNotesProps) => {
-  const { state, setState, reload, setReload, thread, setReplyTo } = props;
-
-  const [isLoading, setIsLoading] = useState(true);
+  const { state, setState, thread, setReplyTo, setReload, patientId, reload } =
+    props;
+  const queryClient = useQueryClient();
 
-  const fetchNotes = async () => {
-    setIsLoading(true);
-    const { data }: any = await request(routes.getPatientNotes, {
-      pathParams: { patientId: props.patientId },
-      query: {
-        offset: (state.cPage - 1) * RESULTS_PER_PAGE_LIMIT,
-        thread,
-      },
-    });
+  const { data, isLoading, fetchNextPage, hasNextPage } = useInfiniteQuery({
+    queryKey: ["notes"],
+    queryFn: async ({ pageParam = 0 }) => {
+      const response = await callApi(routes.getPatientNotes, {
+        pathParams: { patientId },
+        queryParams: { thread, offset: pageParam },
+      });
+      setReload(false);
 
-    if (state.cPage === 1) {
-      setState((prevState: any) => ({
-        ...prevState,
-        notes: data.results,
-        totalPages: Math.ceil(data.count / pageSize),
-      }));
-    } else {
-      setState((prevState: any) => ({
-        ...prevState,
-        notes: [...prevState.notes, ...data.results],
-        totalPages: Math.ceil(data.count / pageSize),
-      }));
-    }
-    setIsLoading(false);
-    setReload(false);
-  };
+      return {
+        results: response?.results ?? [],
+        nextPage: pageParam + RESULTS_PER_PAGE_LIMIT,
+        totalResults: response?.count ?? 0,
+      };
+    },
+    getNextPageParam: (lastPage, allPages) => {
+      const currentResults = allPages.flatMap((page) => page.results).length;
+      if (currentResults < lastPage.totalResults) {
+        return lastPage.nextPage;
+      }
+      return undefined;
+    },
+    initialPageParam: 0,
+  });
 
   useEffect(() => {
-    if (reload || thread) {
-      fetchNotes();
-    }
-  }, [reload, thread]);
+    if (data?.pages) {
+      const allNotes = data.pages.flatMap((page) => page.results);
+
+      const notesMap = new Map(allNotes.map((note) => [note.id, note]));
+
+      const deduplicatedNotes = Array.from(notesMap.values());
 
-  const handleNext = () => {
-    if (state.cPage < state.totalPages) {
       setState((prevState: any) => ({
         ...prevState,
-        cPage: prevState.cPage + 1,
+        notes: deduplicatedNotes,
       }));
-      setReload(true);
     }
-  };
+  }, [data]);
+
+  useEffect(() => {
+    setReload?.(true);
+    queryClient.invalidateQueries({
+      queryKey: ["notes"],
+    });
+  }, [thread]);
```

Comment:
these kind of hacks not needed if the thread name is present in the queryKey itself right?
```suggestion
```